bot/exts/utilities/epoch.py

```diff
@@ -0,0 +1,134 @@
+        """
+        Convert a relative or absolute date/time string to an arrow.Arrow object.
+
+        Try to interpret the date string as a relative time. If conversion fails, try to interpret it as an absolute
+        time. Tokens that are not recognised are returned along with the part of the string that was successfully
+        converted to an arrow object. If the date string cannot be parsed, BadArgument is raised.
+        """
+        try:
+            return arrow.utcnow().dehumanize(argument)
+        except ValueError:
+            try:
+                dt, ignored_tokens = parser.parse(argument, fuzzy_with_tokens=True)
+                return arrow.get(dt), ignored_tokens
+            except parser.ParserError:
+                raise commands.BadArgument(f"`{argument}` Could not be parsed to a relative or absolute date")
+
+
+class Epoch(commands.Cog):
+    """Convert an entered time and date to a unix timestamp."""
+
+    @commands.command(name="epoch")
+    async def epoch(self, ctx: commands.Context, *, date_time: DateString = None) -> None:
+        """
+        Convert an entered date/time string to the equivalent epoch.
+
+        **Relative time**
+            accepted units: "seconds", "minutes", "hours", "days", "weeks", "months", "years"
+            Eg ".epoch in a month 4 days and 2 hours"
+
+        **Absolute time**
+            eg ".epoch 2022/6/15 16:43 -04:00"
+            Absolute times must be entered in descending orders of magnitude.
+            Timezones may take the following forms:
+                Z (UTC)
+                ±HH:MM
+                ±HHMM
+                ±HH
+
+        Times in the dropdown are shown in UTC
+        """
+        if not date_time:
+            await invoke_help_command(ctx)
+            return
+
+        if isinstance(date_time, tuple):
+            ignored_tokens, date_time = list(a.strip() for a in date_time[1] if a.strip()), date_time[0]
+            if ignored_tokens:
+                await ctx.send(f"Could not parse the following token(s): `{', '.join(ignored_tokens)}`")
+                await ctx.send(f"The resulting date and time is: `{date_time.format(arrow.FORMAT_RSS)}`")
+
+        epoch = int(date_time.timestamp())
+        dropdown = _TimestampDropdown(self._format_dates(date_time), epoch)
+        view = _TimestampMenuView(ctx, dropdown)
+        original = await ctx.send(f"`{epoch}`", view=view)
+        if await view.wait():  # wait until expiration and remove the dropdown
+            await original.edit(view=None)
+
+    @staticmethod
+    def _format_dates(date: arrow.Arrow) -> list[str]:
+        """
+        Return a list of date strings formatted according to the discord timestamp styles.
+
+        These are used in the description of each style in the dropdown
+        """
+        date = date.to('utc')
+        formatted = [str(int(date.timestamp()))]
+        formatted += [date.format(format[1]) for format in list(STYLES.values())[1:6]]
```

Comment:
This should be `[1:7]`, currently the Long Date/Time is getting cut-off and Relative Time is being labeled as the Long Date/Time